mountpoint-s3-client/src/s3_crt_client.rs

```diff
@@ -662,6 +718,10 @@ impl ObjectClient for S3CrtClient {
     type PutObjectRequest = S3PutObjectRequest;
     type ClientError = S3RequestError;
 
+    fn part_size(&self) -> Option<usize> {
+        Some(self.inner.part_size)
```

Comment:
Is this the actual part size or just what we set in ClientConfig? AFAIK the CRT will do some clamping.

Reply:
Yeah, the CRT doesn't let us interrogate the actual part size. But right now the only clamping it does is to the 5GiB part size limit, so I added the same clamping to our constructor.